modules/lfs/client.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package lfs
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Client is used to communicate with the LFS server
+type Client struct {
+	client    *http.Client
+	transfers map[string]TransferAdapter
+}
+
+// NewClient creates a LFS client
+func NewClient(hc *http.Client) *Client {
+	client := &Client{hc, make(map[string]TransferAdapter)}
+
+	basic := &BasicTransferAdapter{hc}
+
+	client.transfers[basic.Name()] = basic
+
+	return client
+}
+
+func (c *Client) transferNames() []string {
+	keys := make([]string, len(c.transfers))
+
+	i := 0
+	for k := range c.transfers {
+		keys[i] = k
+		i++
+	}
+
+	return keys
+}
+
+func (c *Client) batch(url, operation string, objects []*Pointer) (*BatchResponse, error) {
```

Comment:
You need to make this cancellable. Pass in a go context.Context to set as the context for the http.Request either here or at definition of the client.